src/full_coverage_path_planner.cpp

```diff
@@ -32,218 +32,217 @@
-}
+    // create a message for the plan
+    nav_msgs::msg::Path gui_path;
+    gui_path.poses.resize(path.size());
 
-void FullCoveragePathPlanner::parsePointlist2Plan(const geometry_msgs::msg::PoseStamped& start,
-    std::list<Point_t> const& goalpoints,
-    std::vector<geometry_msgs::msg::PoseStamped>& plan)
-{
-  geometry_msgs::msg::PoseStamped new_goal;
-  std::list<Point_t>::const_iterator it, it_next, it_prev;
-  int dx_now, dy_now, dx_next, dy_next, move_dir_now = 0, move_dir_next = 0;
-  bool do_publish = false;
-  float orientation = eDirNone;
-  RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Received goalpoints with length: %lu", goalpoints.size());
-  if (goalpoints.size() > 1)
-  {
-    for (it = goalpoints.begin(); it != goalpoints.end(); ++it)
+    if (!path.empty())
     {
-      it_next = it;
-      it_next++;
-      it_prev = it;
-      it_prev--;
+      gui_path.header.frame_id = path[0].header.frame_id;
+      gui_path.header.stamp = path[0].header.stamp;
+    }
 
-      // Check for the direction of movement
-      if (it == goalpoints.begin())
-      {
-        dx_now = it_next->x - it->x;
-        dy_now = it_next->y - it->y;
-      }
-      else
-      {
-        dx_now = it->x - it_prev->x;
-        dy_now = it->y - it_prev->y;
-        dx_next = it_next->x - it->x;
-        dy_next = it_next->y - it->y;
-      }
+    // Extract the plan in world co-ordinates, we assume the path is all in the same frame
+    for (unsigned int i = 0; i < path.size(); i++)
+    {
+      gui_path.poses[i] = path[i];
+    }
 
-      // Calculate direction enum: dx + dy*2 will give a unique number for each of the four possible directions because
-      // of their signs:
-      //  1 +  0*2 =  1
-      //  0 +  1*2 =  2
-      // -1 +  0*2 = -1
-      //  0 + -1*2 = -2
-      move_dir_now = dx_now + dy_now * 2;
-      move_dir_next = dx_next + dy_next * 2;
-
-      // Check if this points needs to be published (i.e. a change of direction or first or last point in list)
-      do_publish = move_dir_next != move_dir_now || it == goalpoints.begin() ||
-                   (it != goalpoints.end() && it == --goalpoints.end());
-
-      // Add to vector if required
-      if (do_publish)
+    plan_pub_->publish(gui_path);
+  }
+
+  void FullCoveragePathPlanner::parsePointlist2Plan(const geometry_msgs::msg::PoseStamped &start,
+                                                    std::list<Point_t> const &goalpoints,
+                                                    std::vector<geometry_msgs::msg::PoseStamped> &plan)
+  {
+    geometry_msgs::msg::PoseStamped new_goal;
+    std::list<Point_t>::const_iterator it, it_next, it_prev;
+    int dx_now, dy_now, dx_next, dy_next, move_dir_now = 0, move_dir_next = 0;
+    bool do_publish = false;
+    float orientation = eDirNone;
+    RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Received goalpoints with length: %lu", goalpoints.size());
+    if (goalpoints.size() > 1)
```

Comment:
I would make this an early-out. So you don't need to indent the whole typical flow for just this condition.
You might even issue a warning then if this is unexpected behavior.